version: '3'

services:
  jupyterhub:
    build: 
      context: ./jupyterhub/jhub_docker_spawner/context
      dockerfile: ../dockerfiles/Dockerfile
    image: jhub_ds #wanglabneuro/jhub_ds:latest
    container_name: jupyterhub
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - jupyterhub_data:/srv/jupyterhub
    environment:
      - DOCKER_JUPYTER_CONTAINER=jlab_matlab:v1
      - DOCKER_NETWORK_NAME=${COMPOSE_PROJECT_NAME}_default
      - HUB_IP=jupyterhub
      - HOST
    labels:
      - "traefik.enable=true"
      - "traefik.frontend.rule=Host:${HOST}"
    restart: on-failure

  jupyterlab:
    build: 
      context: ./jupyterlab/jlab_matlab/context
      dockerfile: ../dockerfiles/Dockerfile
    image: jlab_matlab:v1 #wanglabneuro/jlab_matlab:v1
    container_name: jupyterlab-throwaway
    # volumes:       
    #   - ~/data/d:/data
    #   - ~/data/shared_files:/srv/shared_files
    #   - ~/data/shared_code:/srv/shared_code
    network_mode: none
    command: echo

  reverse-proxy:
    image: traefik:1.7
    container_name: reverse-proxy
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - ./reverse-proxy/traefik.toml:/etc/traefik/traefik.toml
      - ./secret:/etc/certs
      - /var/run/docker.sock:/var/run/docker.sock
    restart: on-failure

volumes:
  jupyterhub_data: